modules/calendar/search.php

```diff
@@ -0,0 +1,173 @@
+            <option value="0">All Providers</option>
+            <?php
+            foreach($providers as $provider) {
+              $id = $provider['id'];
+              $selected = ($_POST['provider_id'] == $id) ? "selected" : "";
+              echo "<option value='$id' $selected>" . htmlspecialchars($provider['lname'],ENT_QUOTES) . ", " . htmlspecialchars($provider['fname'],ENT_QUOTES) . "</option>\n";
+            }
+            ?>
+          </select>
+        </div>
+      </div>
+      <div class="form-group">
+        <label class="control-label col-sm-2" for="facility_id">Facility</label>
+        <div class="col-sm-10">
+          <select class="form-control" id="facility_id" name="facility_id">
+            <option value="0">All Facilities</option>
+            <?php
+            foreach($facilities as $facility) {
+              $id = $facility['id'];
+              $selected = ($_POST['facility_id'] == $id) ? "selected" : "";
+              echo "<option value='$id' $selected>" . htmlspecialchars($facility['name'],ENT_QUOTES) . "</option>\n";
+            }
+            ?>
+          </select>
+        </div>
+      </div>
+      <div class="form-group"> 
+        <div class="col-sm-offset-2 col-sm-10">
+          <button type="submit" class="btn btn-default" name="submit" value="1">Submit</button>
+        </div>
+      </div>
+    </form>
+    
+    <!-- if search performed -->
+    <?php if($_SERVER['REQUEST_METHOD'] == 'POST') { 
+      $pid = ($_POST['provider_id'] == "0") ? null : $_POST['provider_id'];
+      $fid = ($_POST['facility_id'] == "0") ? null : $_POST['facility_id'];
+      $appointments = fetchAppointments(fixDate($_POST['startDate']), fixDate($_POST['endDate']), null, $pid, $fid);
+      
+      // filter the appointments using the keyword
+      $filteredAppointments = array_filter($appointments, "keywordFilter");
+    ?>
+    <table class="table table-hover">
+      <tr>
+        <th>Date</th>
+        <th>Time</th>
+        <th>Provider</th>
+        <th>Patient</th>
+        <th>Category</th>
+      </tr>
+      <?php foreach($filteredAppointments as $appointment) {
+        $eid = $appointment['pc_eid'];
+        $pid = $appointment['pc_pid'];
+        $date = $appointment['pc_eventDate'];
+        echo "<tr class='resultRow' onclick='openEvent(\"$eid\", \"$date\")'>";
+        echo "<td>" . oeFormatShortDate($appointment['pc_eventDate']) . "</td>";
+        echo "<td>" . oeFormatTime($appointment['pc_startTime']) . " - " . oeFormatTime($appointment['pc_endTime']) . "</td>";
+        echo "<td>" . $appointment['ulname'] . ", " . $appointment['ufname'] . "</td>";
+        echo "<td onclick='goPid($pid)' title='Go to Patient'><a href='javascript:;'>" . $appointment['lname'] . ", " . $appointment['fname'] . "</a></td>";
+        echo "<td>" . $appointment['pc_title'] . "</td>";
+        echo "</tr>";
+      }
+      ?>
+    </table>
+    <!-- end if -->
```

Comment:
It would help us old guys later if you would name the loops at the also start as well as the end. Trust me once the code gets large that little information will be helpful something like filter appointments loop start and then the end of the loop